clients/unshaded/src/main/java/tachyon/client/FileOutStream.java

```diff
@@ -228,12 +220,13 @@ public void write(byte[] b, int off, int len) throws IOException {
   public void write(int b) throws IOException {
     if (mWriteType.isCache()) {
       try {
-        if (mCurrentBlockId == -1 || mCurrentBlockLeftByte == 0) {
+        if (mCurrentBlockOutStream == null) {
+          getNextBlock();
+        } else if (mCurrentBlockOutStream.getRemainingSpaceByte() == 0) {
```

Comment:
I am not sure whether java computes conditions in "if" from left to right or from right to left, to make sure there is no NullPointerException, I split the check to if ... else if...